docs/site/tutorials/todo-list/todo-list-tutorial-controller.md

```diff
@@ -182,6 +182,10 @@ $ npm start
 Server is running at http://127.0.0.1:3000
 ```
 
+{% include note.html content="
```

Comment:
I added this because there are some default filter objects that produce zero results or cause an http 500. So clearing these default filter objects provided by the API Explorer (swagger ui?) is best.

Reply:
GET /todos has a default filter of

```
{
  "offset": 0,
  "limit": 100,
  "skip": 0,
  "where": {
    "additionalProp1": {}
  },
  "include": [
    {
      "relation": "string",
      "scope": {
        "offset": 0,
        "limit": 100,
        "skip": 0,
        "where": {
          "additionalProp1": {}
        },
        "include": [
          {
            "additionalProp1": {}
          }
        ]
      }
    }
  ]
}
```

if user presses [Execute] button,  
an exception is thrown

```
{
  "error": {
    "statusCode": 500,
    "message": "Internal Server Error"
  }
}
```

if we clear this filter and try again, the proper number of todos is returned.

Reply:
GET /todos/id

has filter

```
{
  "offset": 0,
  "limit": 100,
  "skip": 0,
  "include": [
    {
      "relation": "string",
      "scope": {
        "offset": 0,
        "limit": 100,
        "skip": 0,
        "where": {
          "additionalProp1": {}
        },
        "include": [
          {
            "additionalProp1": {}
          }
        ]
      }
    }
  ]
}
```
produces 500 error as well. Must be cleared.

Reply:
GET /todos/count 

has a where object

```
{
  "additionalProp1": {}
}
```

that returns 0. Clearing it returns proper amount.

Reply:
GET /todo-lists/{id}/todo-list-image
and
/todo-lists/{id}/todo

have same filter which causes error 500. Clearing them works.

Reply:
perhaps I should change the note.html to

```
{% include note.html content=" When using the API Explorer, be sure to clear out any default filter object or where object as this may affect your results. " %}
```

Reply:
Do we have more details about what causes the 500 error? Any stacktrace?

Reply:
> Do we have more details about what causes the 500 error? Any stacktrace?

GET /todos  (with default filter above) returns:

```
Server is running at http://[::1]:3000
Request GET /todos?filter=%7B%0A%20%20%22offset%22%3A%200%2C%0A%20%20%22limit%22%3A%20100%2C%0A%20%20%22skip%22%3A%200%2C%0A%20%20%22where%22%3A%20%7B%0A%20%20%20%20%22additionalProp1%22%3A%20%7B%7D%0A%20%20%7D%2C%0A%20%20%22include%22%3A%20%5B%0A%20%20%20%20%7B%0A%20%20%20%20%20%20%22relation%22%3A%20%22string%22%2C%0A%20%20%20%20%20%20%22scope%22%3A%20%7B%0A%20%20%20%20%20%20%20%20%22offset%22%3A%200%2C%0A%20%20%20%20%20%20%20%20%22limit%22%3A%20100%2C%0A%20%20%20%20%20%20%20%20%22skip%22%3A%200%2C%0A%20%20%20%20%20%20%20%20%22where%22%3A%20%7B%0A%20%20%20%20%20%20%20%20%20%20%22additionalProp1%22%3A%20%7B%7D%0A%20%20%20%20%20%20%20%20%7D%2C%0A%20%20%20%20%20%20%20%20%22include%22%3A%20%5B%0A%20%20%20%20%20%20%20%20%20%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%22additionalProp1%22%3A%20%7B%7D%0A%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%5D%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%5D%0A%7D failed with status code 500. Error: Invalid "filter.include" entries: {"relation":"string","scope":{"offset":0,"limit":100,"skip":0,"where":{"additionalProp1":{}},"include":[{"additionalProp1":{}}]}}
    at Object.includeRelatedModels (/Users/dremond/Documents/PRs_LoopBack/authTwo/loopback-next/packages/repository/dist/relations/relation.helpers.js:70:21)
    at TodoRepository.includeRelatedModels (/Users/dremond/Documents/PRs_LoopBack/authTwo/loopback-next/packages/repository/dist/repositories/legacy-juggler-bridge.js:389:28)
    at TodoRepository.find (/Users/dremond/Documents/PRs_LoopBack/authTwo/loopback-next/packages/repository/dist/repositories/legacy-juggler-bridge.js:286:21)
    at processTicksAndRejections (internal/process/task_queues.js:94:5)
    at async /Users/dremond/Documents/PRs_LoopBack/authTwo/loopback-next/packages/rest/dist/providers/invoke-method.provider.js:40:32
    at async /Users/dremond/Documents/PRs_LoopBack/authTwo/loopback-next/packages/express/dist/middleware-interceptor.js:82:25
    at async /Users/dremond/Documents/PRs_LoopBack/authTwo/loopback-next/packages/express/dist/middleware-interceptor.js:82:25
    at async /Users/dremond/Documents/PRs_LoopBack/authTwo/loopback-next/packages/rest/dist/providers/send.provider.js:39:32
    at async MiddlewareSequence.handle (/Users/dremond/Documents/PRs_LoopBack/authTwo/loopback-next/packages/rest/dist/sequence.js:201:9)
    at async HttpHandler._handleRequest (/Users/dremond/Documents/PRs_LoopBack/authTwo/loopback-next/packages/rest/dist/http-handler.js:69:9)

```

Reply:
I opened a separate issue https://github.com/strongloop/loopback-next/issues/6641 to track this potential problem in the API Explorer. It isn't related to the re-generated artifacts of this todolist example per se.

Reply:
Same error happens with existing todo-list example

```
Server is running at http://127.0.0.1:3000
Unhandled error in GET /todos?filter=%7B%0A%20%20%22offset%22%3A%200%2C%0A%20%20%22limit%22%3A%20100%2C%0A%20%20%22skip%22%3A%200%2C%0A%20%20%22where%22%3A%20%7B%0A%20%20%20%20%22additionalProp1%22%3A%20%7B%7D%0A%20%20%7D%2C%0A%20%20%22fields%22%3A%20%7B%0A%20%20%20%20%22id%22%3A%20true%2C%0A%20%20%20%20%22title%22%3A%20true%2C%0A%20%20%20%20%22desc%22%3A%20true%2C%0A%20%20%20%20%22isComplete%22%3A%20true%2C%0A%20%20%20%20%22todoListId%22%3A%20true%0A%20%20%7D%2C%0A%20%20%22include%22%3A%20%5B%0A%20%20%20%20%7B%0A%20%20%20%20%20%20%22relation%22%3A%20%22string%22%2C%0A%20%20%20%20%20%20%22scope%22%3A%20%7B%0A%20%20%20%20%20%20%20%20%22offset%22%3A%200%2C%0A%20%20%20%20%20%20%20%20%22limit%22%3A%20100%2C%0A%20%20%20%20%20%20%20%20%22skip%22%3A%200%2C%0A%20%20%20%20%20%20%20%20%22where%22%3A%20%7B%0A%20%20%20%20%20%20%20%20%20%20%22additionalProp1%22%3A%20%7B%7D%0A%20%20%20%20%20%20%20%20%7D%2C%0A%20%20%20%20%20%20%20%20%22fields%22%3A%20%7B%0A%20%20%20%20%20%20%20%20%20%20%22additionalProp1%22%3A%20%7B%7D%0A%20%20%20%20%20%20%20%20%7D%2C%0A%20%20%20%20%20%20%20%20%22include%22%3A%20%5B%0A%20%20%20%20%20%20%20%20%20%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%22additionalProp1%22%3A%20%7B%7D%0A%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%5D%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%5D%0A%7D: 500 Error: Invalid "filter.include" entries: {"relation":"string","scope":{"offset":0,"limit":100,"skip":0,"where":{"additionalProp1":{}},"fields":{"additionalProp1":{}},"include":[{"additionalProp1":{}}]}}
    at Object.includeRelatedModels (/Users/dremond/Documents/PRs_LoopBack/authTwo/loopback4-example-todo-list/node_modules/@loopback/repository/dist/relations/relation.helpers.js:70:21)
    at TodoRepository.includeRelatedModels (/Users/dremond/Documents/PRs_LoopBack/authTwo/loopback4-example-todo-list/node_modules/@loopback/repository/dist/repositories/legacy-juggler-bridge.js:389:28)
    at TodoRepository.find (/Users/dremond/Documents/PRs_LoopBack/authTwo/loopback4-example-todo-list/node_modules/@loopback/repository/dist/repositories/legacy-juggler-bridge.js:286:21)
    at processTicksAndRejections (internal/process/task_queues.js:94:5)
    at async /Users/dremond/Documents/PRs_LoopBack/authTwo/loopback4-example-todo-list/node_modules/@loopback/rest/dist/providers/invoke-method.provider.js:40:32
    at async /Users/dremond/Documents/PRs_LoopBack/authTwo/loopback4-example-todo-list/node_modules/@loopback/express/dist/middleware-interceptor.js:82:25
    at async /Users/dremond/Documents/PRs_LoopBack/authTwo/loopback4-example-todo-list/node_modules/@loopback/express/dist/middleware-interceptor.js:82:25
    at async /Users/dremond/Documents/PRs_LoopBack/authTwo/loopback4-example-todo-list/node_modules/@loopback/rest/dist/providers/send.provider.js:39:32
    at async MiddlewareSequence.handle (/Users/dremond/Documents/PRs_LoopBack/authTwo/loopback4-example-todo-list/node_modules/@loopback/rest/dist/sequence.js:201:9)
    at async HttpHandler._handleRequest (/Users/dremond/Documents/PRs_LoopBack/authTwo/loopback4-example-todo-list/node_modules/@loopback/rest/dist/http-handler.js:69:9)
```

@raymondfeng  ^^